audb/core/repository.py

```diff
@@ -1,6 +1,147 @@
+        it creates an instance of :class:`audbackend.backend.Artifactory`
+        and wraps an :class:`audbackend.interface.Maven` interface
+        around it.
+        The files will then be stored
+        with the following structure on the Artifactory backend
+        (shown by the example of version 1.0.0 of the emodb dataset)::
+
+            emodb/db/1.0.0/db-1.0.0.yaml   <-- header
+            emodb/db/1.0.0/db-1.0.0.zip    <-- dependency table
+            emodb/attachment/.../1.0.0/... <-- attachments
+            emodb/media/.../1.0.0/...      <-- media files
+            emodb/meta/.../1.0.0/...       <-- tables
+
+        When :attr:`Repository.backend` equals ``file-system``,
+        it creates an instance of :class:`audbackend.backend.FileSystem`
+        and wraps an :class:`audbackend.interface.Versioned` interface
+        around it.
+        The files will then be stored
+        with the following structure on the Artifactory backend
+        (shown by the example of version 1.0.0 of the emodb dataset)::
+
+            emodb/1.0.0/db.yaml            <-- header
+            emodb/1.0.0/db.zip             <-- dependency table
+            emodb/attachment/1.0.0/...     <-- attachments
+            emodb/media/1.0.0/...          <-- media files
+            emodb/meta/1.0.0/...           <-- tables
+
+        The returned backend instance
+        has not yet established a connection to the backend.
+        To establish a connection,
+        use the backend with a ``with`` statement,
+        or use the ``open()`` and ``close()`` methods of the backend class.
+        The backend is stored as the inside the ``backend`` attribute
+        of the returned backend interface.
+
+        Returns:
+            interface to repository
+
+        """
+        backend_class = self.backend_registry[self.backend]
+        backend = backend_class(self.host, self.name)
+        if self.backend == "artifactory":
+            interface = audbackend.interface.Maven(backend)
+        else:
+            interface = audbackend.interface.Versioned(backend)
+        return interface
+
+    @classmethod
+    def register(
+        cls,
+        backend_name: str,
+        backend_class: typing.Type[audbackend.backend.Base],
+    ):
+        r"""Register backend class.
+
+        Adds an entry to the dictionary
+        stored in the class variable ``Repository.registry``,
+        mapping a backend name
+        to an actual backend class.
+
+        Args:
+            backend_name: name of the backend,
+                e.g. ``"file-system"``
+            backend_class: class of the backend,
+                that should be associated with ``backend_name``,
+                e.g. ``"audbackend.backend.Filesystem"``
+
+        Examples:
+            >>> import audbackend
+            >>> Repository.register("file-system", audbackend.backend.FileSystem)
+
+        """
+        cls.backend_registry[backend_name] = backend_class
```

Comment:
At the moment we have the class variable `Repository.backend_registry` and the method `Repository.register()` to add a new entry to the registry. I was wondering if this is really needed, or if I just should remove `Repository.register()` as a user/developer can directly change the class variable?

Reply:
@ChristianGeng as you marked this as resolved, should we stay with the current extra `register()` method, or what would be your preferred solution?
I'm tending slightly towards removing `Repository.register()`.

Reply:
I thought I had commented on that - but indeed cannot find it any more either. So apologies in advance in case this should be a reduplication: 

For me it feels a bit awkward to manipulate the `backend_registry ` class variable (and class variables in general) directly. So for me the "setter" feels more natural. In case you would want to maninpulate it directly, could one not think of handling it like  `REPOSITORIES` - which is handled in `config`?

Reply:
OK, I see the point of using a "setter", and will stay with `Repository.register()`.
For now I would not handle it inside `config`, as in order to set another backend a user would first have to implement one. So, there is no real choice at the moment.
At the moment, we use the method in the tests to add custom backends to test certain cases, e.g.

https://github.com/audeering/audb/blob/9dd2abd3aa46a6f7c3e873c59f371df9729aa90f/tests/test_lock_db.py#L34-L45

As I said, I will stay with the current solution, and mark this conversation as resolved.